src/SqlPersistence/SynchronizedStorage/StorageAdapter.cs

```diff
@@ -43,11 +44,21 @@ public Task<CompletableSynchronizedStorageSession> TryAdapt(OutboxTransaction tr
             return session;
         }
 
-        Transaction existingTransaction;
+        Transaction transportTx;
         // Transport supports DTC and uses TxScope owned by the transport
-        if (transportTransaction.TryGet(out existingTransaction))
+        transportTransaction.TryGet(out transportTx);
+        if (transportTx != null && Transaction.Current != null && !transportTx.Equals(Transaction.Current))
+        {
+            throw new Exception("A TransctionScope has been opened in the current context overriding the one created by the transport. " 
+                + "This setup can result in insonsistent data because operations done via connections enlisted in the context scope won't be committed "
+                + "atomically with the receive transaction. If you wish to manually control the TransactionScope in the pipeline switch the transport transaction mode "
+                + $"to values lower than '{nameof(TransportTransactionMode.TransactionScope)}'.");
```

Comment:
shouldnt it be `{nameof(TransportTransactionMode)}.{nameof(TransportTransactionMode.TransactionScope)}` ?